src/rdkafka_lz4.c

```diff
@@ -209,11 +209,11 @@ rd_kafka_lz4_decompress (rd_kafka_broker_t *rkb, int proper_hc, int64_t Offset,
                 goto done;
         }
 
-        /* If uncompressed size is unknown or out of bounds make up a
-         * worst-case uncompressed size
+        /* If uncompressed size is unknown or out of bounds, use a sane
+         * default (2x compression) and reallocate if needed
          * More info on max size: http://stackoverflow.com/a/25751871/1821055 */
         if (fi.contentSize == 0 || fi.contentSize > inlen * 255)
-                estimated_uncompressed_size = inlen * 255;
+                estimated_uncompressed_size = inlen * 2;
```

Comment:
This seems a bit conservative.

Maybe we should make use of of `message.max.bytes` as a starting point?
`estimated_uncompressed_size = RD_MIN(inlen * 255, RD_MAX(inlen * 2, rkb->rkb_rk->rk_conf.max_msg_size));`

Reply:
Sure.